lib/internal/net.js

```diff
@@ -5,7 +5,9 @@ module.exports = { isLegalPort };
 // Check that the port number is not NaN when coerced to a number,
 // is an integer and that it falls within the legal range of port numbers.
 function isLegalPort(port) {
-  if (typeof port === 'string' && port.trim() === '')
+  if (port === null) return true;
```

Comment:
I think if we're monkeying with this function, we should get it to a point where it ONLY accepts valid ports (numbers and numeric strings). Then, if `null` or `undefined` is a valid value for a given calling context, that context can be responsible for allowing those values.